week2/lc1361_validateBinaryTreeNodes.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @param {number} n
+ * @param {number[]} leftChild
+ * @param {number[]} rightChild
+ * @return {boolean}
+ */
+var validateBinaryTreeNodes = function(n, leftChild, rightChild) {
+    let intersect = leftChild.filter((l) => rightChild.includes(l)); // find node that have more than one parents
+    intersect = intersect.filter(i => i !== -1);
+    if (intersect.length !== 0) return false;
+    
+    let levels = new Array(n).fill(0);
+    for (let i = 0; i < leftChild.length; i++) {
+        if (leftChild[i] !== -1) levels[leftChild[i]]++;
+        if (rightChild[i] !== -1) levels[rightChild[i]]++;
+    }
+
+    let ret = levels.filter(l => l === 0); // find level === 0, which means we have single root
+    
```

Comment:
if the binary tree is valid, all number in levels should be 1 except root. So, if we add another filter 
`let ret2 = levels.filter(l => l > 1)
if (ret.length !== 1 || ret2.length > 0) return false;`
instead of filter in the first line of the function, the time can reduce to O(n).

Reply:
Thanks